src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -7717,6 +7720,20 @@ void ObjectMgr::LoadGameObjectTemplateAddons()
             }
         }
 
+        uint32 artKitIndex = 0;
+        for (uint32& artKitID : gameObjectAddon.artKits)
```

Comment:
This code is stupid, you use range loop syntax to avoid having to introduce a separate index variable... but then you have that variable anyway because you need it for logging, just use classic `for (std::size_t i = 0; i < gameObjectAddon.artKits.size(); ...`

Reply:
Also why isnt it just checked in the same loop as assignment from query result?

Reply:
/shrug I just liked having related errors together